src/main/cpp/feedback/DragonLeds.cpp

```diff
@@ -53,125 +67,160 @@ void DragonLeds::setOff()
 {
     m_addressibleLeds->Stop();
 }
-
-DragonLeds::DragonLeds() : m_addressibleLeds()
+void DragonLeds::ResetVariables()
 {
+    m_loopThroughIndividualLEDs = -1;
+    m_colorLoop = 0;
+    m_timer = 0;
+    m_switchColor = false;
 }
 
-DragonLeds *DragonLeds::m_instance = nullptr;
-DragonLeds *DragonLeds::GetInstance()
+void DragonLeds::commitLedData()
 {
-    if (m_instance == nullptr)
+    if (m_ledBuffer.size() > 0)
     {
-        m_instance = new DragonLeds();
+        m_addressibleLeds->SetData(m_ledBuffer);
     }
-    return m_instance;
+}
+void DragonLeds::SetSolidColor(frc::Color colar)
+{
+    frc::LEDPattern pattern = frc::LEDPattern::Solid(colar);
+    pattern.ApplyTo(m_ledBuffer);
 }
 
-std::array<int, 3> DragonLeds::getColorValues(Colors c)
+void DragonLeds::SetAlternatingColor(frc::Color colar1, frc::Color colar2)
 {
-    switch (c)
+    for (unsigned int i = 0; i < m_ledBuffer.size(); i++)
     {
-    case RED:
-        return {255, 0, 0};
-    case GREEN:
-        return {0, 255, 0};
-    case BLUE:
-        return {0, 0, 255};
-    case YELLOW:
-        return {255, 160, 0};
-    case PURPLE:
-        return {75, 0, 130};
-    case AZUL:
-        return {0, 255, 255};
-    case WHITE:
-        return {255, 255, 180};
-    case BLACK:
-        return {0, 0, 0};
-    default:
-        return {0, 0, 0};
+        if (i % 2 == 0)
+            m_ledBuffer[i].SetLED(colar1);
```

Comment:
I know you like alternate spellings bug what is the story behind colar tha I am missing